iroh-docs/src/actor.rs

```diff
@@ -581,7 +581,10 @@ impl Drop for SyncHandle {
     fn drop(&mut self) {
         // this means we're dropping the last reference
         if let Some(handle) = Arc::get_mut(&mut self.join_handle) {
-            self.tx.send(Action::Shutdown { reply: None }).ok();
+            // this call is the reason tx can not be a tokio mpsc channel.
+            // we have no control about where drop is called, yet tokio send_blocking panics
+            // when called from inside a tokio runtime.
+            self.tx.send_blocking(Action::Shutdown { reply: None }).ok();
```

Comment:
couldn't we detect the current runtime, and if one exists use `block_on` + `send` and otherwise use `send_blocking`?

Reply:
I don't think so. block_on panics if you call it from inside a runtime... You could use spawn(send), but really I just don't want to do anything runtime related during a Drop...

```
Panics
This function panics if the provided future panics, if called within an asynchronous execution context, or if a timer future is executed on a runtime that has been shut down.
```